vim_menutrans_helper.py

```diff
@@ -0,0 +1,196 @@
+                if len(tobe_translated_blocks) > 0 and tobe_translated_blocks[0] == '5':
+                    tobe_translated_blocks = tobe_translated_blocks[1:]
+                if len(tobe_translated_blocks) > 0 and tobe_translated_blocks[-1] == '5':
+                    tobe_translated_blocks = tobe_translated_blocks[:-1]
+
+                # menu_item with "ToolBar" constructed a tobe_translated_blocks with length equals zero
+                if len(tobe_translated_blocks) == 0:
+                    continue
+
+                # untranslated word string splitted by "."
+                tobe_translated_word = tobe_translated_blocks.split('.')
+                # untranslated word string splitted by "." but with "\." together
+                tobe_translated_words = []
+
+                # fill tobe_translated_words with '\.' together words
+                pre_backlash = False
+                for word in tobe_translated_word:
+                    if len(word) != 0 and word[0] == '-':
+                        pre_backlash = False
+                        continue
+                    if pre_backlash:
+                        tobe_translated_words[-1] += '.' + word
+                    else:
+                        tobe_translated_words.append(word)
+                    if len(word) != 0 and word[-1] == '\\':
+                        pre_backlash = True
+                    else:
+                        pre_backlash = False
+
+                # conver tobe_translated_words list to string
+                for word in tobe_translated_words:
+                    if len(word) > 0:
+                        untranslated_dict[word.lower()] = (
+                            line_number, untranslated_file)
+            elif word_list[0] in toolbar_item:
+                untranslated_dict[word_list[1].lower()] = (
+                    line_number, untranslated_file)
+            elif word_list[0] == "let" and (len(word_list) > 1 and word_list[1].startswith("g:menutrans_")):
+                untranslated_dict[word_list[1].lower()] = (
+                    line_number, untranslated_file)
+
+    # PopUp doesn't need to be translated
+    untranslated_dict.pop("popup", None)
+
+    # blank with digit doesn't neet to be translated
+    temp_untranslated_dict = {}
+    for u in untranslated_dict.keys():
+        if (u.startswith('\\ ') and u.endswith('\\ ')) or (len(u) == 1 and u[0] >= '0' and u[0] <= '9'):
+            temp_untranslated_dict[u] = untranslated_dict[u]
+
+    for key in temp_untranslated_dict.keys():
+        if key in untranslated_dict:
+            del untranslated_dict[key]
+
+
+def makeTranslatedDic(translated_file, translated_dict):
+    translated_menu = {"tmenu", "menut", "menutrans", "menutranslate"}
+    # deal with the translated file
+    with open(translated_file, encoding='utf-8') as f2:
+        for line_number, line in enumerate(f2):
+            # list without '\blank' together per line
+            word_list = line.split()
```

Comment:
Don't split the file then re-join to handle the escape characters. Use regex all the way through.